packages/compartment-mapper/src/types.js

```diff
@@ -190,6 +190,7 @@ export const moduleJSDocTypes = true;
  * @param {string} specifier
  * @param {string} location
  * @param {string} packageLocation
+ * @param {ReadFn | ReadPowers} [readPowers]
```

Comment:
I believe you’re proposing that we add `fileURLToPath` to `ReadPowers`.

  I’m okay with `importLocation` using `fileURLToPath` but not okay with `makeArchive` or `parseArchive` using them because the behavior is platform-dependent, these functions must be consistent regardless of platform, and the values would not be useful for archives. For backward-compatibility, the new function would need to be optional and have a fallback. Since this is only used for `__filename` and `__dirname` in CommonJS, I’m okay with the fallback being to provide `null` and `null`. Providing a fallback that gives right answers only on some platforms would be sketchy.